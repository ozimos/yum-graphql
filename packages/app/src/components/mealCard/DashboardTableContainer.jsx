import React from 'react';
import PropTypes from 'prop-types';
import ReactTable from 'react-table';
import format from 'date-fns/format';
import 'react-table/react-table.css';

const DashboardTableContainer = ({ orders, ...props }) => {

  const columns = [
    {
      Header: 'Order Id',
      accessor: 'id',
      className: 'test-click',
      width: 300,
    }, {
      Header: 'Customer',
      accessor: 'User',
      Cell: columnProps => (
        <span>
          {columnProps.row.User.firstName} &nbsp;
          {columnProps.row.User.lastName}
        </span>
      ),
      width: 150,
    }, {
      Header: 'Customer Id',
      accessor: 'userId',
      width: 200,
    }, {
      Header: 'Date',
      accessor: 'updatedAt',
      Cell: columnProps =>
        (<span>{format(columnProps.value, 'DD MMM, YYYY')}</span>),
      width: 100,
    }, {
      Header: 'Time',
      accessor: 'updatedAt',
      Cell: columnProps => (<span>{format(columnProps.value, 'h:mm A')}</span>),
      width: 100
    }
  ];
  return (
    <div>
      <ReactTable
        data={orders}
        columns={columns}
        className="-striped -highlight clickable"
        minRows={0}
        defaultPageSize={props.pagination.limit}
        loading={props.loading}
        pages={props.pagination.pages}
        onFetchData={props.onFetchData}
        sortable={false}
        manual
        getTdProps={(state, rowInfo, column) => ({
            onClick: () => {
              if (column.Header) {
                props.getOrderMeals(rowInfo.row.id);
                props.getOrderMealsTotals(rowInfo.row.id);
              }
            }
          })}
      />
    </div>
  );
};

DashboardTableContainer.defaultProps = {
  pagination: {
    pages: 1,
    limit: 10,
    offset: 0
  },
  loading: false,
};
DashboardTableContainer.propTypes = {
  orders: PropTypes.arrayOf(PropTypes.object).isRequired,
  getOrderMeals: PropTypes.func.isRequired,
  getOrderMealsTotals: PropTypes.func.isRequired,
  onFetchData: PropTypes.func.isRequired,
  pagination: PropTypes.shape({
    pages: PropTypes.number,
    limit: PropTypes.number,
    offset: PropTypes.number,
  }),
  loading: PropTypes.bool
};

export default DashboardTableContainer;

