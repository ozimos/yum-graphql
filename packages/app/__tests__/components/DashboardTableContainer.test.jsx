/* global React:false, shallow:false, toJson:false */

import DashboardTableContainer from
  '../../src/components/mealCard/DashboardTableContainer';
import { allOrders } from '../__mocks__/orderDataMock';

const props = {
  pagination: {
    limit: 10,
    offset: 0,
    pages: 1
  },
  currentOrderId: 'abc',
  loading: false,
  orders: allOrders.data.rows,
  getOrderMeals: jest.fn(),
  getOrderMealsTotals: jest.fn(),
  onFetchData: jest.fn(),
};

describe('DashboardTableContainer Component', () => {

  const setup = () => shallow(<DashboardTableContainer {...props} />);

  it('should render without throwing an error', () => {
    const wrapper = setup();
    expect(wrapper
      .exists(<table className="table" />)).toBe(true);
  });

  it('renders correctly', () => {
    const wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

