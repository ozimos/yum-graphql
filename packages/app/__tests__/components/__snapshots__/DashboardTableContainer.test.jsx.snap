// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardTableContainer Component renders correctly 1`] = `
<div>
  <ReactTable
    AggregatedComponent={[Function]}
    ExpanderComponent={[Function]}
    FilterComponent={[Function]}
    LoadingComponent={[Function]}
    NoDataComponent={[Function]}
    PadRowComponent={[Function]}
    PaginationComponent={[Function]}
    PivotValueComponent={[Function]}
    ResizerComponent={[Function]}
    TableComponent={[Function]}
    TbodyComponent={[Function]}
    TdComponent={[Function]}
    TfootComponent={[Function]}
    ThComponent={[Function]}
    TheadComponent={[Function]}
    TrComponent={[Function]}
    TrGroupComponent={[Function]}
    aggregatedKey="_aggregated"
    className="-striped -highlight clickable"
    collapseOnDataChange={true}
    collapseOnPageChange={true}
    collapseOnSortingChange={true}
    column={
      Object {
        "Aggregated": undefined,
        "Cell": undefined,
        "Expander": undefined,
        "Filter": undefined,
        "Footer": undefined,
        "Header": undefined,
        "Pivot": undefined,
        "PivotValue": undefined,
        "aggregate": undefined,
        "className": "",
        "filterAll": false,
        "filterMethod": undefined,
        "filterable": undefined,
        "footerClassName": "",
        "footerStyle": Object {},
        "getFooterProps": [Function],
        "getHeaderProps": [Function],
        "getProps": [Function],
        "headerClassName": "",
        "headerStyle": Object {},
        "minWidth": 100,
        "resizable": undefined,
        "show": true,
        "sortMethod": undefined,
        "sortable": undefined,
        "style": Object {},
      }
    }
    columns={
      Array [
        Object {
          "Header": "Order Id",
          "accessor": "id",
          "className": "test-click",
          "width": 300,
        },
        Object {
          "Cell": [Function],
          "Header": "Customer",
          "accessor": "User",
          "width": 150,
        },
        Object {
          "Header": "Customer Id",
          "accessor": "userId",
          "width": 200,
        },
        Object {
          "Cell": [Function],
          "Header": "Date",
          "accessor": "updatedAt",
          "width": 100,
        },
        Object {
          "Cell": [Function],
          "Header": "Time",
          "accessor": "updatedAt",
          "width": 100,
        },
      ]
    }
    data={
      Array [
        Object {
          "MealsURL": "url",
          "User": Object {
            "email": "asdf",
            "firstName": "first",
            "lastName": "last",
          },
          "id": "order1",
          "updatedAt": "2018-08-24T16:47:55.783Z",
          "userId": "userId",
        },
        Object {
          "MealsURL": "url",
          "User": Object {
            "email": "asdf",
            "firstName": "first",
            "lastName": "last",
          },
          "id": "order2",
          "updatedAt": "2018-08-24T16:47:55.783Z",
          "userId": "userId",
        },
        Object {
          "MealsURL": "url",
          "User": Object {
            "email": "asdf",
            "firstName": "first",
            "lastName": "last",
          },
          "id": "order3",
          "updatedAt": "2018-08-24T16:47:55.783Z",
          "userId": "userId",
        },
      ]
    }
    defaultExpanded={Object {}}
    defaultFilterMethod={[Function]}
    defaultFiltered={Array []}
    defaultPageSize={10}
    defaultResized={Array []}
    defaultSortDesc={false}
    defaultSortMethod={[Function]}
    defaultSorted={Array []}
    expanderDefaults={
      Object {
        "filterable": false,
        "resizable": false,
        "sortable": false,
        "width": 35,
      }
    }
    filterable={false}
    freezeWhenExpanded={false}
    getLoadingProps={[Function]}
    getNoDataProps={[Function]}
    getPaginationProps={[Function]}
    getProps={[Function]}
    getResizerProps={[Function]}
    getTableProps={[Function]}
    getTbodyProps={[Function]}
    getTdProps={[Function]}
    getTfootProps={[Function]}
    getTfootTdProps={[Function]}
    getTfootTrProps={[Function]}
    getTheadFilterProps={[Function]}
    getTheadFilterThProps={[Function]}
    getTheadFilterTrProps={[Function]}
    getTheadGroupProps={[Function]}
    getTheadGroupThProps={[Function]}
    getTheadGroupTrProps={[Function]}
    getTheadProps={[Function]}
    getTheadThProps={[Function]}
    getTheadTrProps={[Function]}
    getTrGroupProps={[Function]}
    getTrProps={[Function]}
    groupedByPivotKey="_groupedByPivot"
    indexKey="_index"
    loading={false}
    loadingText="Loading..."
    manual={true}
    minRows={0}
    multiSort={true}
    nestingLevelKey="_nestingLevel"
    nextText="Next"
    noDataText="No rows found"
    ofText="of"
    onFetchData={[MockFunction]}
    originalKey="_original"
    pageSizeOptions={
      Array [
        5,
        10,
        20,
        25,
        50,
        100,
      ]
    }
    pageText="Page"
    pages={1}
    pivotDefaults={Object {}}
    pivotIDKey="_pivotID"
    pivotValKey="_pivotVal"
    previousText="Previous"
    resizable={true}
    resolveData={[Function]}
    rowsText="rows"
    showPageJump={true}
    showPageSizeOptions={true}
    showPagination={true}
    showPaginationBottom={true}
    showPaginationTop={false}
    sortable={false}
    style={Object {}}
    subRowsKey="_subRows"
  />
</div>
`;
